articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,453 @@
+        }
+```
+
+However, this code alone can't be executed by Q#.
+For that, it needs to make up the body of an [operation](xref:microsoft.quantum.guide.basics#q-operations-and-functions), which is then executed when called---either directly or by another operation. 
+Hence, you can write an operation of the following form:
+```qsharp
+    operation MeasureSuperposition() : Result {
+        using (q = Qubit()) {
+            H(q);
+            return MResetZ(q);
+        }
+    }
+```
+You have defined an operation, `MeasureSuperposition`, which takes no inputs and returns a value of type [Result](xref:microsoft.quantum.guide.types).
+More details on defining callables can be found at [Operations and functions](xref:microsoft.quantum.guide.operationsfunctions).
+For the sake of brevity we refer primarily to Q# *operations* on this page, however all of the concepts discussed work in the same manner for Q# *functions*, as well as any user-defined Q# types. 
+
+### Operation defined in a Q# file
+
+The operation is precisely what's called and run by Q#.
+However, it requires a few more additions to comprise a full `*.qs` Q# file.
+
+All Q# types and callables are defined within *namespaces*, which the compiler can then open individually 
+
+Firstly, the [`H`](xref:microsoft.quantum.intrinsic.h) and [`MResetZ`](xref:microsoft.quantum.measurement.mresetz) operations actually belong to the [Q# Standard Libraries](xref:microsoft.quantum.qsharplibintro).
+To be available for use, their respective *namespaces* need to be opened.
+All Q# callables and types (both those you define and those intrinsic to the language) are defined within namespaces.
+To access them, the compiler needs to be made aware of which namespaces to open.
+
+So, the full Q# file containing our operation consists of defining our own namespace, opening the namespaces for the operations our operation uses, and then our operation:
+
+```qsharp
+namespace NamespaceName {
+    open Microsoft.Quantum.Intrinsic;     // for the H operation
+    open Microsoft.Quantum.Measurement;   // for MResetZ
+
+    operation MeasureSuperposition() : Result {
+        using (q = Qubit()) { 
+            H(q);
+            return MResetZ(q);
+        }
+    }
+}
+```
+
+### Execution of a Q# program
+
+Now the general execution model of a Q# program becomes clear.
+
+<br/>
+<img src="./images/general_execution_model.png" alt="Q# program execution diagram" width="400">
+
+Firstly, the specific operation to be executed has access to any other callables and types defined in the same namespace.
+Thanks to `open` statements, it can also access those from any of the [Q# libraries](xref:microsoft.quantum.libraries). 
+
+The operation itself is then executed on a *[target machine](xref:microsoft.quantum.machines)*.
+In the future, one such target machine will be a real quantum computer, but there are currently multiple simulators available, each with a particular use.
```

Comment:
I think we can stop saying "in the future" and "real" as we are able to provide quantum hardware as target machines on Azure Quantum now as private preview. 
 ```suggestion
Such target machine can be quantum hardware, and multiple simulators available, each with a particular use.
```